src/providers/Onboarding.js

```diff
@@ -85,6 +94,24 @@ function OnboardingProvider({ children }) {
     []
   )
 
+  const getTransactions = useCallback(async () => {
+    // Token approvals
+    const txs = [...createTokenApproveTxs(config)]
+
+    // Tx one
+    txs.push(createGardenTxOne(config))
+
+    if (config.garden.type === NATIVE_TYPE) {
+      // Mint seeds balances
+      txs.push(createTokenHoldersTx(config))
+    }
+
+    // Tx two, tx three
+    txs.push(createGardenTxTwo(config), await createGardenTxThree(config))
```

Comment:
Yes i think that it makes sense!

If we go with the `react-query` route, i think the best approach would be to upload the file when the transaction steps screen is rendered and pass the covenant ipfs hash to the `getTransactions` function. What do you think?

Reply:
Yes, I like that!

Reply:
hey guys in this case i would need to argue to keep adding libraries to the repo with things that could  easily be  managed manually, we have bunch of examples about retries that we implemented

Reply:
take a look to the size of the bundle <https://bundlephobia.com/package/react-query@3.21.1> and this quote from a convo with pierre some time ago :
``` But in general, I try to keep every module to the minimum, so that we don’t end up with an app being 1MB+
My mental target is always ~100kb for a complete app… which is really hard to achieve
But that’s how I evaluate modules to add, I ask myself if I’m ok to give e.g. 10% -20% of the size of this ideal app size to a certain library ```